src/views/catalog/CreateFromInstanceTypes/components/AddBootableVolumeModal/components/VolumeMetadata/components/InstanceTypeDrilldownSelect/components/ComposableDrilldownSelect/ComposableDrilldownSelect.tsx

```diff
@@ -0,0 +1,127 @@
+import React, {
+  Dispatch,
+  FC,
+  KeyboardEvent,
+  ReactNode,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+
+import { Menu, MenuContent, MenuList, MenuToggle, Popper } from '@patternfly/react-core';
+
+type MenuHeightsType = {
+  [id: string]: number;
+};
+
+type ComposableDrilldownMenuProps = {
+  toggleLabel: ReactNode | string;
+  isOpen: boolean;
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
+  scrollableMenuIDs?: string[];
+  isScrollable?: boolean;
+  id?: string;
+};
+
+const ComposableDrilldownSelect: FC<ComposableDrilldownMenuProps> = ({
+  children,
+  toggleLabel,
+  isOpen,
+  setIsOpen,
+  scrollableMenuIDs = [],
+  isScrollable = false,
+  id = 'rootMenu',
+}) => {
+  const [activeMenu, setActiveMenu] = useState<string>(id);
+  const [menuDrilledIn, setMenuDrilledIn] = useState<string[]>([]);
+  const [drilldownPath, setDrilldownPath] = useState<string[]>([]);
+  const [menuHeights, setMenuHeights] = useState<MenuHeightsType>({});
+
+  const handleClickOutside = useCallback(
+    (event: MouseEvent) => {
+      if (isOpen && !document.getElementById(id)?.contains(event.target as Node)) {
+        setIsOpen(false);
+      }
+    },
+    [id, isOpen, setIsOpen],
+  );
+
+  const onToggleClick = (ev?: React.MouseEvent) => {
+    ev?.stopPropagation(); // Stop handleClickOutside from handling
+    setIsOpen(!isOpen);
+    setMenuDrilledIn([]);
+    setDrilldownPath([]);
+    setActiveMenu(id);
+  };
+
+  const drillIn = (
+    _event: KeyboardEvent | React.MouseEvent,
+    fromMenuId: string,
+    toMenuId: string,
+    pathId: string,
+  ) => {
+    setMenuDrilledIn([...menuDrilledIn, fromMenuId]);
+    setDrilldownPath([...drilldownPath, pathId]);
+    setActiveMenu(toMenuId);
+  };
+
+  const drillOut = (_event: KeyboardEvent | React.MouseEvent, toMenuId: string) => {
+    setMenuDrilledIn(menuDrilledIn.slice(0, menuDrilledIn.length - 1));
+    setDrilldownPath(drilldownPath.slice(0, drilldownPath.length - 1));
+    setActiveMenu(toMenuId);
+  };
+
+  const setHeight = (menuId: string, height: number) => {
+    if (!menuHeights[menuId] || (menuId !== id && menuHeights[menuId] !== height)) {
+      setMenuHeights((prev) => ({
+        ...prev,
+        [menuId]: height,
+      }));
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('click', handleClickOutside);
+    return () => {
+      window.removeEventListener('click', handleClickOutside);
+    };
+  }, [handleClickOutside, isOpen]);
+
+  const toggle = (
```

Comment:
why nested? u can. simply use where needed, its a regular const, no benefit in using as this

Reply:
the only reason is that how it showed in PF example and I was too lazy to change that lol, changing now :)